Products/PluggableAuthService/PluggableAuthService.py

```diff
@@ -437,15 +437,21 @@ def _setObject(self, id, object, roles=None, user=None, set_owner=0):
         Folder._setObject(self, id, object, roles, user, set_owner)
 
     security.declarePrivate('_delOb')
-
     def _delOb(self, id):
         #
-        #   Override ObjectManager's version to clean up any plugin
-        #   registrations for the deleted object
+        # Override ObjectManager's version to clean up any plugin
+        # registrations for the deleted object
         #
+        # For ZopeVersionControl, we need to check 'plugins' for more than
+        # existence, since it replaces objects (like 'plugins') with 
+        # SimpleItems and calls _delOb, which tries to use special methods
+        # of 'plugins'
+
+        # XXX imo this is a evil one
+
         plugins = self._getOb('plugins', None)
 
-        if plugins is not None:
+        if getattr(plugins, 'removePluginById', None) is not None:
```

Comment:
Why are we modifying this code here?  ZVC shouldn't be replacing a PAS instance's `plugins` with something which does not implement `Products.PluginRegistry.interfaces.IPluginRegistry`.

Reply:
@esteele Can you please explain the reason for this change in more detail as you committed it in the first place?

Reply:
I just pulled it from the PlonePAS patch. That change is at least 11 years old. 
https://github.com/plone/Products.PlonePAS/blame/1d8f9263b2ab6df93d67d5fbf66d49527c57f57a/src/Products/PlonePAS/pas.py#L374

Reply:
@esteele Was it ever used in a release or in production or can we drop this PR as irrelevant?

Reply:
The patch in PlonePAS has been in use for the past eleven years, and is happily used by any non-stone-age Plone site. Merging the patch upstream in PluggableAuthService would mean we can remove a patch from PlonePAS.
This particular check seems harmless.

Reply:
I concur with Tres. Why can't those things that ZVC uses as ``plugins`` grow a ``removePluginById`` method? Special cases like that don't really belong upstream.